tracer/test/test-applications/debugger/dependency-libs/Samples.Probes.TestRuns/SmokeTests/AsyncRecursiveCallWithMultipleProbes.cs

```diff
@@ -0,0 +1,29 @@
+using System.Runtime.CompilerServices;
+using System.Threading.Tasks;
+
+namespace Samples.Probes.TestRuns.SmokeTests
+{
+    public class AsyncRecursiveCallWithMultipleProbes : IAsyncRun
+    {
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        public async Task RunAsync()
+        {
+            await Recursive(0);
+        }
+
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        [LogMethodProbeTestData(expectedNumberOfSnapshots: 3)]
```

Comment:
1. Have you tested the scenario where we have multiple async method probes but each has a different message template?
2. It would be good to test the multi-probe scenario (both async and non-async) with the the "phase" concept, to make sure we can add 1 more probe, remove 1 of the probes (or all the probes), etc, and get the expected results.

Reply:
1. The only problem is that we scrub out the message template. I discussed it with Dudi in the past we should maybe revive this discussion.
2. Agree. I'll add phasing 👍

Reply:
Added in 64307fdca251d02ac719aab38a903ba902181d7b.